CARDS-2797: Enforce various useful ESLint rules

veronikaslc opened the pull request:
Report error for unused vars

veronikaslc committed:
CARDS-2797: Enforce various useful ESLint rules
Report error for unused vars

sdumitriu commented on modules/data-entry/src/main/frontend/src/questionnaire/Note.jsx:
```diff
@@ -40,7 +40,6 @@ function Note (props) {
     pageActive,
     fullSize,
     value,
-    onAddSuggestion,
```
This was removed before (590d61278d3dbaf94879bdc692ee182d1c1cc7fb) and reverted (03e28eecee1f81503681ed373aff29e677deb4ad) because it introduces a JS error. There are two versions of the `Note` component, one that requires this prop, and one that doesn't need it, but it is passed down as a prop either way in case it is needed. This needs testing on a question with notes enabled.

sdumitriu replied:
I just checked, it does indeed trigger the JS error. It's in here because otherwise it is included in `rest` and passed to `<TextField>`. Maybe we shouldn't pass `rest` to `TextField`, let's investigate which props are relevant and pass only those explicitly (if they aren't too many).

veronikaslc replied:
`rest` has to stay, we transit `noteProps` via it that are defined in dicom question

sdumitriu commented on modules/data-entry/src/main/frontend/src/questionnaire/SelectableAreaQuestion.jsx:
```diff
@@ -53,7 +53,7 @@ import FormattedText from "../components/FormattedText.jsx";
 function SelectableAreaQuestion(props) {
   checkPropTypes(SelectableAreaQuestion, props);
-  let { classes, errorText, existingAnswer, questionName, questionDefinition, pageActive, isEdit, ...rest } = props;
+  let { classes, existingAnswer, questionName, questionDefinition, pageActive, isEdit, ...rest } = props;
```
To check: I think this was being pulled out of `rest` so it doesn't get passed down to `Answer`.

sdumitriu commented on modules/data-entry/src/main/frontend/src/vocabQuery/VocabularyBrowser.jsx:
```diff
@@ -86,7 +86,7 @@ function VocabularyBrowser(props) {
     }
-
+    // eslint-disable-next-line no-unused-vars
     for (const [key, value] of Object.entries(browserOpened ? buttonRefs : infoButtonRefs)) {
```
This could instead be changed to `for (const [, value] of` to avoid an unused var.

sdumitriu commented on modules/data-entry/src/main/frontend/src/dataHomepage/Filters.jsx:
```diff
@@ -194,6 +194,7 @@ function Filters(props) {
 
     // Load up the comparators for this index, if not already loaded
+    // eslint-disable-next-line no-unused-vars
     let [loadedComparators, component] = getOutputChoices(path);
```
If `component` is not used, this could just be:


```suggestion
    let [loadedComparators] = getOutputChoices(path);
```

veronikaslc committed:
CARDS-2797: Enforce various useful ESLint rules
Optimised unused errors report, reverted removal of onAddSuggestion in the Notes component

sdumitriu commented on modules/data-entry/src/main/frontend/src/dataHomepage/Filters.jsx:
```diff
@@ -194,7 +194,7 @@ function Filters(props) {
 
     // Load up the comparators for this index, if not already loaded
-    let [loadedComparators, component] = getOutputChoices(path);
+    let [loadedComparators] = getOutputChoices(path);
```
Unrelated issue, we seem to call `getOutputChoices` again just a few lines later, is that a bug caused by various edits and refactorings, or is the second call really needed?
